### PR TITLE
fix: Updating shape props to undefined when using editor.updateShape

### PR DESCRIPTION
**Pull request by kazu-2020**
When using `editor.store.put`, any property within the props of a shape can be updated to undefined. However, when using `editor.updateShape`, updates to undefined are ignored.

To address this, I have modified `editor.updateShape` to allow updating any property within props to undefined as well.

related: #4953
### Change type

- [x] `bugfix`
- [ ] `improvement`
- [ ] `feature`
- [ ] `api`
- [ ] `other`

### Test plan

- [x] Unit tests
- [ ] End to end tests

### Release notes

- Updating shape props to undefined  when using editor.updateShape

### CODE REVIEW

**Comment by kazu-2020** [comment]
I have read and agree to the Contributor License Agreement.

**Comment by steveruizok** [comment]
I think we could land this! I'll wait for another approval, just in case there is something I'm missing. (FWIW I'd recommend using `null` rather than `undefined` for "optional" properties).

**Comment by kazu-2020** [comment]
> I'd recommend using `null` rather than `undefined` for "optional" properties

Yes, that is what I thought when I read the code...

My question is, if I replace the props of the persisted data in this case from undefined to null, is using [tldraw's migration](https://tldraw.dev/docs/persistence#Shape-props-migrations) function a better approach? 🤔

**Review by SomeHats — APPROVED** [review]
lgtm!

**Comment by steveruizok** [comment]
Pushing an empty commit to tell the robots it's cool

**Comment by trygve-aaberge-adsk** [comment]
Hm, this is a pretty hidden breaking change. If someone has a conditional property they send to `updateShapes` which they set to a value if they want the prop updated and to undefined if they don't want it updated, it will now be set to undefined instead. This set will not be caught by type checking unless `exactOptionalPropertyTypes` is enabled (which is default off) and it will bypass the prop validation so you may end up with a property set to undefined which does not allow the value to be undefined. This in turn may make the app crash when trying to read it.

I don't think this has affected us, but having this released in a patch version (3.6.1) without any notes about it when it has the potential to destroy the data in shapes is pretty bad imo.

I can see that it's useful to be able to set a property to undefined, but having different behavior when you specify a property as undefined and when you not specify it seems a bit surprising to me.

**Comment by mimecuvalo** [comment]
You make a great point @trygve-aaberge-adsk . I think it was a slip-up that that change ended up in 3.6.1. It looks like it was meant for the 3.7.0 release (and was noted in those docs). 

I think we'll have to discuss this more internally, you do raise some good points on the DX on this. Seems like a footgun for folks...

**Comment by trygve-aaberge-adsk** [comment]
> You make a great point @trygve-aaberge-adsk . I think it was a slip-up that that change ended up in 3.6.1. It looks like it was meant for the 3.7.0 release (and was noted in those docs).

Yeah, good that the change was mentioned in 3.7.0, however I think it would be better if it was also mentioned that it was a breaking change that could have this impact.

> I think we'll have to discuss this more internally, you do raise some good points on the DX on this. Seems like a footgun for folks...

Thanks, great if you can discuss if there's a better way to look out for and handle changes like this. I feel like breaking changes, especially ones like this that can easily be missed and have a bad impact, are important to look out for and communicate clearly.

**Pull request merged**
